x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/edit/index.tsx

```diff
@@ -98,6 +100,11 @@ const EditRulePageComponent: FC = () => {
   const [ruleLoading, rule] = useRule(ruleId);
   const loading = ruleLoading || userInfoLoading || listsConfigLoading;
 
+  const { isSavedQueryLoading, savedQueryBar, savedQuery } = useGetSavedQuery(rule?.saved_id, {
+    ruleType: rule?.type,
+    onError: noop,
```

Comment:
`onError: noop,` reduces straightness of the code since it requires knowledge what's going on under the hood and suppressing of that behaviour when necessary. I'd recommend to consider removing the default error handling functionality inside, exposing of `showErrorToastHandler` and passing it directly when necessary. Though it uses `toasts` then it makes sense to expose a hook like `useShowErrorToastHandler()` which can be composable and reusable.

Reply:
@maximpn 
Toast is not displayed in this case because error is displayed on edit form when saved query is not loaded. Thus it's prevents showing duplicated notifications

<img width="1046" alt="Screenshot 2022-10-10 at 11 58 27" src="https://user-images.githubusercontent.com/92328789/194851564-fe249f6b-d95c-4f21-85cf-83dc34ce120e.png">